packages/build/src/plugins/node_version.ts

```diff
@@ -78,6 +98,37 @@ const addPluginNodeVersion = function ({
     'https://answers.netlify.com/t/build-plugin-update-system-node-js-version-upgrade-to-20/108633',
   )}`,
     )
+
+    if (pluginPath) {
+      const pluginDir = dirname(pluginPath)
+      const { packageJson: pluginPackageJson } = await getPackageJson(pluginDir)
+
+      // Ensure Node.js version is compatible with plugin's `engines.node`
+      const pluginNodeVersionRange = pluginPackageJson?.engines?.node
+      if (!pluginNodeVersionRange) {
+        logWarning(
+          logs,
+          "This plugin doesn't have a `package.json` file with an `engines.node` field, so it might not work well with Node.js 20.",
+        )
+        systemLog(`plugin "${packageName}" might be affected by node.js 20 change`)
+      } else if (semver.satisfies('20.0.0', pluginNodeVersionRange)) {
+        logWarning(
+          logs,
+          'In its package.json, the plugin claims to be compatible with Node.js 20, so this upgrade should go alright.',
+        )
```

Comment:
should we really warn in this case?

Reply:
idk, up to you. I care about being able to query for this in our logs, so the system log should be enoug.

Reply:
removed the user-facing log in https://github.com/netlify/build/pull/5497/commits/7b7720580224fa90dd8dcc39fff6d4b4c40a1291